File: Source/Back-end/src/main/java/org/elis/manutenzione/dto/response/GetUtentiResponse.java
package org.elis.manutenzione.dto.response;

import lombok.Data;
import org.elis.manutenzione.model._enum.Ruolo;

import java.time.LocalDate;

/**
 * DTO che rappresenta la risposta per la richiesta di ottenere i dettagli degli utenti.
 */
@Data
public class GetUtentiResponse {

    /**
     * L'indirizzo email dell'utente.
     */
    private String email;

    /**
     * Il nome dell'utente.
     */
    private String nome;

    /**
     * Il cognome dell'utente.
     */
    private String cognome;

    /**
     * Il ruolo dell'utente.
     */
    private Ruolo ruolo;

    /**
     * La data di nascita dell'utente.
     */
    private LocalDate dataDiNascita;
}

File: Source/Front-end/front-end/src/app/model/response/get-utenti-response.ts
import {Ruolo} from "../models/ruolo";

export class GetUtentiResponse {
  email: string;
  nome: string;
  cognome: string;
  ruolo: Ruolo;
  dataDiNascita: Date;

  constructor(email: string, nome: string, cognome: string, dataDiNascita: Date, ruolo: Ruolo) {
    this.email = email;
    this.nome = nome;
    this.cognome = cognome;
    this.ruolo = ruolo;
    this.dataDiNascita = dataDiNascita;

  }
}
